README.md

```diff
@@ -1,23 +1,277 @@
 
-## Examples
+ 1. [Introduction](#introduction)
+ 2. [Performance](#performance)
+ 3. [Getting Started](#getting-started)
+    1. [Publish Subscribe](#publish-subscribe)
+    2. [Events](#events)
+ 4. [Supported Platforms](#supported-platforms)
+ 5. [Language Bindings](#language-bindings)
+ 6. [Thanks To All Contributors](#thanks-to-all-contributors)
 
-### Getting Started
+## Introduction
+
+Welcome to Elkodon, your gateway to fast, efficient, and ultra-low latency
+inter-process communication. This library is the embodiment of Rust's power and
+elegance, designed to meet your communication needs with zero-copy and lock-free
+mechanisms.
+
+Elkodon is all about providing a seamless experience for inter-process
+communication, featuring versatile messaging patterns. Whether you're diving
+into publish-subscribe, events, or the promise of upcoming features like
+request-response, pipelines, and blackboard, Elkodon has you covered.
+
+One of the standout features of Elkodon is its remarkable consistency.
+Regardless of payload size, transmission latency remains constant, ensuring a
+predictable and reliable communication experience.
+
+Elkodon's origins can be traced back to Iceoryx, but it's not just an
+evolution – it's a revolution. By overcoming past technical debts and refining
+the architecture, Elkodon enables the modularity we've always desired.
+
+In the near future, Elkodon is poised to support at least the same feature set
+and platforms as Iceoryx, ensuring a seamless transition and offering enhanced
+capabilities for your inter-process communication needs. So, if you're looking
+for lightning-fast, cross-platform communication that doesn't compromise on
+performance or modularity, Elkodon is your answer. Welcome to a new era of
+inter-process communication.
+
+## Performance
+
+```mermaid
+gantt
+    title Latency (in ns) - 64b payload
+    dateFormat X
+    axisFormat %s
+
+    section elkodon
+    240 : 0, 240
+    section iceoryx
+    1000 : 0, 1000
+    section MQueue
+    700 : 0, 700
+    section UDS
+    1500 : 0, 1500
+```
+
+```mermaid
+gantt
+    title Latency (in ns) - 64kb payload
+    dateFormat X
+    axisFormat %s
+
+    section elkodon
+    240 : 0, 240
+    section iceoryx
+    1000 : 0, 1000
+    section MQueue
+    14000 : 0, 14000
+    section UDS
+    23000 : 0, 23000
+```
+
+**Benchmark-System**
+
+- **CPU:** Intel(R) Core(TM) i7-10875H CPU @ 2.30GHz
+- **OS:** Linux 6.5.9-arch2-1 #1 SMP PREEMPT_DYNAMIC GNU/Linux
+- **Compiler:**
+  - rustc 1.72.1
+  - gcc 13.2.1 20230801
+
+## Getting Started
 
 ### Publish Subscribe
 
+This minimal example showcases a publisher sending the number 1234 every second,
+while a subscriber efficiently receives and prints the data.
+
+**publisher.rs**
+
+```rust
+use elkodon::prelude::*;
+use elkodon_bb_posix::signal::SignalHandler;
+
+fn main() {
+    let service_name = ServiceName::new(b"My/Funk/ServiceName").unwrap();
+
+    let service = zero_copy::Service::new(&service_name)
```

Comment:
Mmmmhhh, in the end it is always IPC but it may makes sense to rename process local into single address space. So if you have a operating system with one big address space, it is the thing for you ... or for process local communication.

So, it may makes sense to rename it into
* `process_local` -> `single_address_space`
* `zero_copy` -> `posix`
* upcoming other variants could `qemu`, `hypervisor` and `vulkan`

Maybe there also comes a copy variant, then I would call it somehow like `posix_non_zero_copy` or so.

@elBoberido What do you think?

Reply:
I thinks it moves into the right direction but still does not yet feel completely right. Have to take council with my pillow.